### ISSUE TITLE
'Select For Update With RS' Cannot be Parsed

### ISSUE THREAD

**Issue by rvrooman-codelogic** [JSQLParser/JSqlParser#1149]
**Describe the bug**
net.sf.jsqlparser.JSQLParserException: Encountered unexpected token: "with" "WITH"

**To Reproduce**

  ```
@Test
public void testJsqlParserExceptionWhenSelectForUpdate() throws JSQLParserException {
    String sqlStatement =
        "select "
          + "test0_.id as id1 "
        + "from test_table test0_ "
        + "where test0_.id=? for update with rs";

    net.sf.jsqlparser.statement.Statement statement = CCJSqlParserUtil.parse(sqlStatement);
  }
```
**Expected behavior**
The example is a 'select for update' acquiring an 'RS' (read stability) lock.

I am not quite sure where this information would go in the current Statement model,
but at a minimum I would expect it not to throw the exception for valid SQL.

**System**
- java 11
- Found with apache derby as DB and hibernate generated query.
- JSQLParser version 4.0

**Comment by wumpz** [comment]
**for update** is already supported. The problem here is **with rs**.

**Comment by rvrooman-codelogic** [comment]
I've updated the title to be more specific. Thanks.

**Comment by manticore-projects** [comment]
This has been fixed and is working with JSQLParser 4.6 Snapshot.
